staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -94,22 +94,23 @@ func (s basicSchemaValidator) ValidateUpdate(new, old interface{}, options ...Va
 
 // NewSchemaValidator creates an openapi schema validator for the given CRD validation using environment.DefaultCompatibilityVersion().
 func NewSchemaValidator(customResourceValidation *apiextensions.JSONSchemaProps) (SchemaValidator, *spec.Schema, error) {
-	return NewSchemaValidatorForVersion(customResourceValidation, environment.DefaultCompatibilityVersion())
+	return NewSchemaValidatorForVersion(customResourceValidation, compatibility.KubeComponentEffectiveVersion().EmulationVersion())
```

Comment:
Need to change the godoc, at least

What are the implications of changing the version this uses by default? It looks like this is used when writing CRDs to decide if all the validation stuff in the CRD spec is valid to persist. Don't we *have* to construct that with our compatibility version to make sure we don't persist a CRD a skewed older API server would choke on?

Reply:
I've switched `NewSchemaValidator` to be fully backward compatible and only support formats available in ALL versions.